payments/voucher_manager.go

```diff
@@ -79,43 +79,45 @@ func (vm *VoucherManager) Pay(channelId types.Destination, amount *big.Int, pk [
 	return voucher, nil
 }
 
-// Receive validates the incoming voucher, and returns the total amount received so far
-func (vm *VoucherManager) Receive(voucher Voucher) (*big.Int, error) {
+// Receive validates the incoming voucher, and returns the total amount received so far as well as the amount received from the voucher
+func (vm *VoucherManager) Receive(voucher Voucher) (total *big.Int, fromVoucher *big.Int, err error) {
 	vInfo, err := vm.store.GetVoucherInfo(voucher.ChannelId)
 	if err != nil {
-		return &big.Int{}, fmt.Errorf("channel not registered: %w", err)
+		return &big.Int{}, &big.Int{}, fmt.Errorf("channel not registered: %w", err)
 	}
 
 	// We only care about vouchers when we are the recipient of the payment
 	if vInfo.ChannelPayee != vm.me {
-		return &big.Int{}, nil
+		return &big.Int{}, &big.Int{}, nil
 	}
-	received := &big.Int{}
-	received.Set(voucher.Amount)
-	if types.Gt(received, vInfo.StartingBalance) {
-		return &big.Int{}, fmt.Errorf("channel has insufficient funds")
+
+	if types.Gt(voucher.Amount, vInfo.StartingBalance) {
+		return &big.Int{}, &big.Int{}, fmt.Errorf("channel has insufficient funds")
```

Comment:
Genuine Q: is there some value in returning the _actual_ total each time, even in the case of an error? It might make error handling less necessary for consumers...

Reply:
I don't think we can always return the `total` since we could fail to fetch the voucher info on L84, and never be able to figure out the current total. To keep things consistent I think it makes sense to not return the total here.